frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx

```diff
@@ -36,17 +57,46 @@ function VariableItem({
 
 				const response = await query({
 					query: variableData.queryValue || '',
+					variables: existingVariables,
 				});
 
 				setIsLoading(false);
 				if (response.error) {
-					setErrorMessage(response.error);
+					let message = response.error;
```

Comment:
can we use `useQuery` hook here to fetch the data over mount

Reply:
I tried doing this. This made it harder to follow the code. Can you share how you would do it?